controllers/common/condition/controller.go

```diff
@@ -139,5 +102,68 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		r.Recorder.Eventf(obj, "Normal", "Failed", "Failed to update conditions: %s", updateError.Error())
 		return ctrl.Result{}, nil
 	}
+
 	return ctrl.Result{}, nil
 }
+
+func diffConditions(obj v1alpha1.InnerObject) (newConditionMap map[v1alpha1.ChaosConditionType]StatusAndReason) {
+	records := obj.GetStatus().Experiment.Records
+	newConditionMap = make(map[v1alpha1.ChaosConditionType]StatusAndReason)
+
+	if records != nil {
+		newConditionMap[v1alpha1.ConditionSelected] = StatusAndReason{
+			Status: corev1.ConditionTrue,
+		}
+	} else {
+		newConditionMap[v1alpha1.ConditionSelected] = StatusAndReason{
+			Status: corev1.ConditionFalse,
+		}
+	}
+
+	// If records is `nil`, we don't need to check the `allInjected` and `allRecovered` conditions.
+	var allInjected corev1.ConditionStatus
```

Comment:
```
allInjected := corev1.ConditionFalse
if records != nil && every(records, func(record *v1alpha1.Record) bool {
	return record.Phase == v1alpha1.Injected
}) {
	allInjected = corev1.ConditionTrue
}
```
I prefer this style instead of `if xxx else xxx`

Reply:
Go style refer this doc: https://github.com/uber-go/guide/blob/master/style.md#unnecessary-else

Reply:
Fixed in [8681aea](https://github.com/chaos-mesh/chaos-mesh/pull/4199/commits/8681aea547c6161d6e72a60184e87ebfe4680499).